cicecore/cicedynB/general/ice_forcing.F90

```diff
@@ -5819,6 +5833,191 @@ subroutine init_snowtable
 
       end subroutine init_snowtable
 
+!=======================================================================
+
+! snicar snow grain single-scattering properties lookup table
+!
+!
+     subroutine init_snicarssptable
+
+      use ice_broadcast, only: broadcast_array, broadcast_scalar
+      integer (kind=int_kind) :: &
+         nsnwgrad, &   ! Table dimensions
+         nspecint
+      real (kind=dbl_kind), allocatable :: &
+         extcffdrc(:,:), &  ! snow mass extinction cross section (m2/kg)
+         extcffdfs(:,:), &  ! snow mass extinction cross section (m2/kg)
+         sscalbdrc(:,:), &  ! snow single scatter albedo (fraction)
+         sscalbdfs(:,:), &  ! snow single scatter albedo (fraction)
+         asmprmdrc(:,:), &  ! snow asymmetry factor (cos(theta))
+         asmprmdfs(:,:)     ! snow asymmetry factor (cos(theta))
+
+      ! local variables
+
+      logical (kind=log_kind) :: diag = .false.
+
+      integer (kind=int_kind) :: &
+         fid                  ! file id for netCDF file
+
+      character (char_len) :: &
+         fieldname            ! field name in netcdf file
+
+      character(len=*), parameter :: subname = '(init_snicarssptable)'
+
+      !-----------------------------------------------------------------
+      ! read table of snicar SSP parameters
+      !-----------------------------------------------------------------
+
+      if (my_task == master_task) then
+         write (nu_diag,*) ' '
+         write (nu_diag,*) 'SNICAR SSP file:', trim(ssp_filename)
+      endif
+
+      if (ssp_filename == 'unknown') then
```

Comment:
shouldn't the hardwired tables be in icepack, not CICE (or other drivers)?

Reply:
This is a good question and a good idea.  I agree we should move the 5x5 default table into Icepack.  There are some technical challenges because those arrays need to be allocated, so there is no way to set the default size/values at compile time.  That means we need to allocate and fill them at run time, but there is no required "icepack init" routine that we can put that in.  What we'll probably have to do is to have a flag/method that checks whether the SSP table data is set and if not, initialize it to the 5x5 data before it is used.  That adds a little extra complexity, but is probably OK.

Separately, we should discuss process.  Do you want me to PR Icepack changes to a SNICAR development branch somewhere?  Or should I PR directly to the E3SM Icepack dev branch?  At the same time I do some development in Icepack, I will update this PR to match.

Reply:
Re process, maybe it depends on what's being merged?  E.g. it makes sense to me for these changes to be PR'd into my snicar branch rather than the E3SM Icepack dev branch, but other things you do to Icepack (not directly snicar-related) could go directly to the E3SM branch. I'm a little ambivalent about that but leaning toward merging closely related work into the same branch so that it can be refined before being merged into the E3SM branch, in case other people pull from the E3SM branch, they won't have to deal with code that might still be changing.  But if your changes are more general, then merge it directly?  Does that make sense?